File: blood-donation/src/app/model/appointment.ts
import { MedicalCenterDTO } from "./medicalCenterDTO";
import { Staff } from "./staff";
import { User } from "./User";

export interface Appointment {
    id: number;
    date: Date;
    user: User;
    medicalCenterDTO: MedicalCenterDTO;
    taken: boolean;
    staff: Staff[];
}

File: blood-donation-app/src/main/java/blood/donation/app/model/Appointment.java
package blood.donation.app.model;

import com.fasterxml.jackson.annotation.JsonBackReference;
import com.fasterxml.jackson.annotation.JsonView;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import org.springframework.lang.Nullable;

import javax.persistence.*;
import java.util.Date;
import java.util.List;

@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Entity
public class Appointment {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    @Column
    private Date date;
    @ManyToOne(fetch = FetchType.EAGER)
    @JoinColumn(name = "user_id")
    private User user;
    @OneToOne
    private MedicalCenter medicalCenter;
    @Column
    private boolean taken = false;
    @ManyToMany
    private List<Staff> staff;
}
